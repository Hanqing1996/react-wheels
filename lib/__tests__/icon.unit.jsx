import * as renderer from "react-test-renderer";
import React from 'react'
import Icon from '../components/icon/icon'
import {mount} from 'enzyme'

describe('icon', () => {
    it('render successfully', () => {
        const json = renderer.create(<Icon name='draw'/>).toJSON()
        expect(json).toMatchSnapshot()
    })
    it('onClick', function () {
        const fn=jest.fn()

        const component=mount(<Icon name='draw' onClick={fn}/>)
        component.find('svg').simulate('click')
        expect(fn).toBeCalled()
    });

})