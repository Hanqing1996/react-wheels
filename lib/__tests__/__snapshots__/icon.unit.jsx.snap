// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon render successfully 1`] = `
<div>
  <svg
    className="icon"
  >
    <use
      xlinkHref="draw"
    />
  </svg>
</div>
`;
